mlflow/utils/autologging_utils/safety.py

```diff
@@ -268,6 +268,9 @@ def is_testing():
     return os.environ.get(_AUTOLOGGING_TEST_MODE_ENV_VAR, "false") == "true"
 
 
+_AUTOLOGGING_INTEGRATIONS_TEMP_DISABLED_SET = set()
```

Comment:
@WeichenXu123 As an alternative to defining a new set of temporarily disabled integrations, could we add a new `state` property to `AutologgingSession` (https://github.com/mlflow/mlflow/blob/94581a3fe4debd2deb9a53b53470c89d69291c29/mlflow/utils/autologging_utils/safety.py#L539) that we can update to `failed` when a failure occurs? We can then then check the state of the active session here: https://github.com/mlflow/mlflow/pull/4276/files#diff-7cea9a64941de9f0572df5ad42d7eef22d2915c2408bb0e59406fe2df20a8167R370.

The reason I prefer this approach is that it scopes the behavior to a single autologging session, rather than creating new global state. This way, we don't require a `try/finally` to ensure that an entry is always removed from this global list.

Let me know what your thoughts are here, @WeichenXu123

Reply:
Regardless of the chosen approach, can we also introduce a test case ensuring that this nesting + failure scenario is handled properly?

Reply:
emm, add AutologgingSession state property doesn't work.
because each new nested autologging call will create a new AutologgingSession, we cannot fetch the parent AutologgingSession.
otherwise we change AutologgingSession to something similar to the _TrainingSession

Reply:
add test, but keep current fixing approach due to reason above.

Reply:
@WeichenXu123 The current implementation doesn't create a new session every time (see https://github.com/mlflow/mlflow/blob/06c8015385d6f1e8c9e1f455e117b99bdbf55773/mlflow/utils/autologging_utils/safety.py#L549-L553)

Reply:
my fault. your approach works. Updated code.